crates/bridge-pot-currency-swap/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! Bridge pot currency swap implementation.
+
+// Either generate code at stadard mode, or `no_std`, based on the `std` feature presence.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    sp_runtime::DispatchError,
+    traits::{Currency, ExistenceRequirement, Get, Imbalance, WithdrawReasons},
+};
+use primitives_currency_swap::CurrencySwap;
+use sp_std::marker::PhantomData;
+
+/// The config for the generic bridge pot currency swap logic.
+pub trait Config<AccountIdFrom, AccountIdTo> {
+    /// The pot account used to receive withdrawed balances to.
+    type PotFrom: Get<AccountIdFrom>;
+
+    /// The pot account used to send deposited balances from.
+    type PotTo: Get<AccountIdTo>;
+}
+
+/// One to one bridge pot currency swap logic.
+pub struct OneToOne<AccountIdFrom, AccountIdTo, ConfigT, CurrencyFrom, CurrencyTo>(
+    PhantomData<(
+        AccountIdFrom,
+        AccountIdTo,
+        ConfigT,
+        CurrencyFrom,
+        CurrencyTo,
+    )>,
+)
```

Comment:
I think we could move all those params to `Config` for better UX. We already have it, so why don't just move the rest of the params (`AccountIdFrom`, `AccountIdTo`, `CurrencyFrom`, `CurrencyTo`) there.